### PR TITLE
Adding support for SHA3 in the transit backend.

### PR DESCRIPTION
**Pull request by robison**
This PR is intended to add support for SHA3 as a hashing algorithm in the `transit` backend. Feedback most appreciated!

### CODE REVIEW

**Comment by victorr** [comment]
Hello @robison!

I'm going to review your PR, but would you mind rebasing it in the meanwhile?

Kind regards.

File: vault/logical_system.go [hunk 1]
```diff
@@ -3291,6 +3293,14 @@ func (b *SystemBackend) pathHashWrite(ctx context.Context, req *logical.Request,
 		hf = sha512.New384()
 	case "sha2-512":
 		hf = sha512.New()
+	case "sha3-224":
```
Comment (victorr) [inline vault/logical_system.go:3296]: What about updating the documentation for this operation in `vault/logical_system_paths.go`?

**Review by victorr — CHANGES_REQUESTED** [review]
The PR looks good to me. Once the unit tests for sign and verify are added the PR will be ready to merge. 

Thank you for the work you have already put into this PR!

**Comment by schultz-is** [comment]
Howdy @robison. Just wanted to let you know that we brought this functionality in with #13367, which includes your commit. Thanks for your contribution!

**Pull request closed without merging**